descartes_moveit/src/ikfast_moveit_state_adapter.cpp

```diff
@@ -151,10 +151,25 @@ void descartes_moveit::IkFastMoveitStateAdapter::setState(const moveit::core::Ro
 bool descartes_moveit::IkFastMoveitStateAdapter::computeIKFastTransforms()
 {
   // look up the IKFast base and tool frame
-  ros::NodeHandle nh;
+  ros::NodeHandle nh(group_name_);
+  std::string ikfast_base_frame_param_key, ikfast_tool_frame_param_key;
   std::string ikfast_base_frame, ikfast_tool_frame;
-  nh.param<std::string>("ikfast_base_frame", ikfast_base_frame, default_base_frame);
-  nh.param<std::string>("ikfast_tool_frame", ikfast_tool_frame, default_tool_frame);
+  if (nh.searchParam("ikfast_base_frame", ikfast_base_frame_param_key))
+  {
+    nh.getParam(ikfast_base_frame_param_key, ikfast_base_frame);
+  }
+  else
+  {
+    ikfast_base_frame = default_base_frame;
```

Comment:
This fallback situation is what can cause race conditions, so it looks like sending a warn message is in place.

Reply:
Not quite related to the race conditions we discussed yesterday, but I think warning the user when using the default parameters does indeed make sense.